proxy/mutli_validator.go

```diff
@@ -0,0 +1,42 @@
+package proxy
+
+import (
+	"context"
+	"encoding/asn1"
+
+	"cvm-reverse-proxy/internal/atls"
+)
+
+// Validator for Azure confidential VM attestation using TDX which accepts multiple measurements
+type MultiValidator struct {
+	oid        asn1.ObjectIdentifier
+	validators []atls.Validator
+}
+
+// NewMultiValidator returns a new Validator for Azure confidential VM attestation using TDX which accepts multiple measurements
+func NewMultiValidator(validators []atls.Validator) *MultiValidator {
+	for _, v := range validators {
+		if !v.OID().Equal(validators[0].OID()) {
+			// This is not an error! This is a bug!
+			panic("validators with mismatching OIDs passed in!")
```

Comment:
Would returning an err be better here?

Reply:
it's fine with me in this case :)

curious why tho. shouldn't generally libraries or imported code never panic but always return an error, because that allows the application to decide what to show and how to handle?

Reply:
Yeah there's something to what you are saying.
In this specific scenario this is more a part of the application rather than a library